src/websites/services/domains.service.ts

```diff
@@ -143,78 +174,145 @@ export class DomainsService extends createPrismaBase(MODELS.Domain) {
+          state: contact.state,
+          zipCode: contact.zipCode,
+        },
+        domain.price.toNumber(),
+      )
 
-    // TODO: how to handle for polling status on server side?
+      const projectResult = await this.vercel.addDomainToProject(domain.name)
 
-    return operationId
+      await this.model.update({
+        where: { id: domain.id },
+        data: {
+          operationId: `vercel-${domain.name}-${Date.now()}`,
+          status: DomainStatus.submitted,
+        },
+      })
+
+      return {
+        vercelResult,
+        projectResult,
+        message: 'Domain registration completed with Vercel',
+      }
+    } catch (error) {
+      this.logger.error('Error registering domain with Vercel:', error)
+
+      await this.model.update({
+        where: { id: domain.id },
+        data: { status: DomainStatus.inactive },
+      })
+
+      throw new BadRequestException(
+        `Failed to register domain with Vercel: ${
+          error instanceof Error ? error.message : 'Unknown error'
+        }`,
+      )
+    }
   }
 
   async configureDomain(websiteId: number) {
     const domain = await this.findUniqueOrThrow({
       where: { websiteId },
     })
 
-    // can only turn off auto renew after registration
-    await this.route53.disableAutoRenew(domain.name)
+    try {
+      const verifyResult = await this.vercel.verifyProjectDomain(domain.name)
+      this.logger.debug('Domain verification result:', verifyResult)
 
-    const route53Response = await this.route53.setDnsRecords(
-      domain.name,
-      RRType.A,
-      VERCEL_DNS_IP, // point to Vercel's Anycast IP
-    )
-    this.logger.debug('Updated domain DNS record', route53Response.ChangeInfo)
+      await this.model.update({
+        where: { id: domain.id },
+        data: { status: DomainStatus.registered },
+      })
 
-    const vercelResponse = await this.vercel.addDomainToProject(domain.name)
-    this.logger.debug('Added domain to Vercel project', vercelResponse)
+      return {
+        domain: domain.name,
+        verified: verifyResult,
+        status: 'configured',
+        message: 'Domain configured successfully with Vercel',
+      }
+    } catch (error) {
+      this.logger.error('Error configuring domain:', error)
+      throw new BadRequestException(
+        `Failed to configure domain: ${
+          error instanceof Error ? error.message : 'Unknown error'
+        }`,
+      )
+    }
+  }
 
-    if (!vercelResponse.verified) {
+  async getPaymentStatus(paymentId: string): Promise<PaymentStatus | null> {
+    try {
+      const paymentIntent = await this.payments.retrievePayment(paymentId)
+      return paymentIntent.status as PaymentStatus
+    } catch (error) {
       this.logger.warn(
-        `Domain ${domain.name} added to Vercel but requires verification`,
-        vercelResponse.verification,
+        `Failed to retrieve payment status for ${paymentId}:`,
+        error,
       )
+      return null
```

Comment:
We want this to fail silently?

Shouldn't a call to this w/ a bad `paymentId` return a 404, or at least a BadGateway? 🤔